### PR TITLE
servlets

### PR DESCRIPTION
**Pull request by MaksymZymyn**

### CODE REVIEW

**Review by sirius91191 — APPROVED** [review]

File: src/main/java/App.java [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 public class App {
     public static void main(String[] args) throws Exception {
+        DatabaseSetup.migrate(HerokuEnv.jdbc_url(), HerokuEnv.jdbc_username(), HerokuEnv.jdbc_password());
+
         Server server = new Server(HerokuEnv.port());
 
         ServletContextHandler handler = new ServletContextHandler();
 
         handler.addServlet(new ServletHolder(new StaticFileServlet("static")), "/static/*");
+        handler.addServlet(new ServletHolder(new StaticFileServlet("static")), "/css/*");
```
Comment (yalukaiwo) [inline src/main/java/App.java:19]: we don't need another endpoint for css, it's included by /static

File: src/main/java/likes/Like.java [hunk 2]
```diff
@@ -1,38 +1,18 @@
...
-    }
-
-    public UUID getUser_to() {
-        return user_to;
-    }
-
-    public boolean isValue() {
-        return value;
-    }
+    UUID id;
```
Comment (yalukaiwo) [inline src/main/java/likes/Like.java:12]: private final ?

File: src/main/java/likes/LikeService.java [hunk 3]
```diff
@@ -1,28 +1,27 @@
 package likes;
 
+import lombok.*;
 import java.util.*;
 
+@AllArgsConstructor
+@Data
 public class LikeService {
-    private final LikeDAO likeDAO;
+    LikeDAO likeDAO;
```
Comment (yalukaiwo) [inline src/main/java/likes/LikeService.java:9]: private final ?

File: src/main/java/servlets/LikeServlet.java [hunk 4]
```diff
@@ -62,18 +57,26 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
...
+                                Auth.renderUnregistered(resp);
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
+                            }
+                        }
                 );
     }
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
```
Comment (yalukaiwo) [inline src/main/java/servlets/LikeServlet.java:76]: what exactly does this doPost do ?

File: src/main/java/servlets/LoginServlet.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                                Auth.renderUnregistered(resp);
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
+                            }
+                        }
+                );
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
```
Comment (yalukaiwo) [inline src/main/java/servlets/LoginServlet.java:48]: // i will rewrite this later

File: src/main/java/users/User.java [hunk 6]
```diff
@@ -1,27 +1,34 @@
...
 public class User {
-    private final UUID id;
-    private final String username;
-    private final String fullName;
-    private String picture;
-    private final String password;
-
-    User(UUID id, String username, String fullName, String picture, String password) {
-        this.id = id;
+    UUID id;
```
Comment (yalukaiwo) [inline src/main/java/users/User.java:14]: private final ?

**Review by yalukaiwo — CHANGES_REQUESTED** [review]

File: src/main/java/servlets/UserServlet.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import java.sql.SQLException;
+import java.util.*;
+
+@Data
+public class UserServlet extends HttpServlet {
+    UserService userService;
+    LikeService likeService;
+    FreemarkerService freemarker;
+    private static int counter;
+    public static ArrayList<User> users = new ArrayList<>();
```
Comment (yalukaiwo) [inline src/main/java/servlets/UserServlet.java:21]: UserServlet should not contain any static variables. Every time LikeServlet needs a list of users, it should resort to corresponding function in UserService.java. same applies for counter. i don't really get the point of this at the moment anyway.

**Review by yalukaiwo — CHANGES_REQUESTED** [review]

**Review by sirius91191 — APPROVED** [review]

**Review by yalukaiwo — APPROVED** [review]
approved after changes

**Pull request merged**